src/lib.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Affero General Public License as published by the
+// Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+// details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>
+
+mod attribute;
+pub use attribute::AttributeElementExt;
+mod only_child;
+pub use only_child::OnlyChildElementExt;
+
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to find attribute '{1}' in element '{0}'")]
+    AttributeNotFound(String, String),
+    #[error("No child with expected criteria in Element '{0}'")]
+    NoChildWithCriteria(String),
```

Comment:
Is `criteria` plural? Otherwise use "criteri`on`.
If so, please do a ctrl+F `criteria` on this PR for all changes :wink:

Reply:
I'd prefer to keep `filter` everywhere because that's the naming already used in [`Iterator` for example](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter). I've changed `Criteria` and `criteria` everywhere in ab7a10bb22110faeb93bf983e0cabf3a223511d9.